### PR TITLE
allows easier creation of many of these modules

### PR DESCRIPTION
**Pull request by Benbentwo**
## what
 * Allows the module to be recreated many times using the same Vault or IAM role.

## why
* We should be able to create many Backup plans that share the same **Vault** and or share the same **IAM Role**
* We should be able to disable components

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -68,22 +68,27 @@ data "aws_iam_policy_document" "assume_role" {
 }
 
 resource "aws_iam_role" "default" {
-  count              = module.this.enabled ? 1 : 0
-  name               = module.label_backup_role.id
+  count              = module.this.enabled && var.iam_enabled ? 1 : 0
```
Comment (osterman) [inline main.tf:71]:
```suggestion
  count              = module.this.enabled && var.iam_role_enabled ? 1 : 0
```

File: main.tf [hunk 2]
```diff
@@ -68,22 +68,27 @@ data "aws_iam_policy_document" "assume_role" {
...
+  name               = var.target_iam_name == "" ? module.label_backup_role.id : var.target_iam_name
```
Comment (osterman) [inline main.tf:72]:
```suggestion
  name               = var.target_iam_role_name == "" ? module.label_backup_role.id : var.target_iam_role_name
```

File: main.tf [hunk 3]
```diff
@@ -68,22 +68,27 @@ data "aws_iam_policy_document" "assume_role" {
...
 }
 
+data "aws_iam_role" "existing" {
+  count = module.this.enabled && var.iam_enabled ? 0 : 1
+  name  = module.label_backup_role.id
+}
+
 resource "aws_iam_role_policy_attachment" "default" {
-  count      = module.this.enabled ? 1 : 0
+  count      = module.this.enabled && var.iam_enabled ? 1 : 0
```
Comment (osterman) [inline main.tf:83]: Suggest using a local instead and reusing the local.

File: variables.tf [hunk 4]
```diff
@@ -67,3 +67,39 @@ variable "copy_action_delete_after" {
...
+  default     = true
+}
+
+variable "iam_enabled" {
+  type        = bool
+  description = "Should we create a new Iam Role and Policy Attachment"
+  default     = true
+}
+
+variable "target_iam_name" {
```
Comment (osterman) [inline variables.tf:101]:
```suggestion
variable "target_iam_role_name" {
```

File: variables.tf [hunk 5]
```diff
@@ -67,3 +67,39 @@ variable "copy_action_delete_after" {
...
+  default     = true
+}
+
+variable "plan_enabled" {
+  type        = bool
+  description = "Should we create a new Plan"
+  default     = true
+}
+
+variable "iam_enabled" {
```
Comment (osterman) [inline variables.tf:95]:
```suggestion
variable "iam_role_enabled" {
```

File: variables.tf [hunk 6]
```diff
@@ -67,3 +67,39 @@ variable "copy_action_delete_after" {
...
+  type        = bool
+  description = "Should we create a new Iam Role and Policy Attachment"
+  default     = true
+}
+
+variable "target_iam_name" {
+  type        = string
+  description = "Override target IAM Name"
+  default     = ""
```
Comment (osterman) [inline variables.tf:104]: Use null instead?

File: variables.tf [hunk 7]
```diff
@@ -67,3 +67,39 @@ variable "copy_action_delete_after" {
...
+variable "plan_name_suffix" {
+  type        = string
+  description = "The string appended to the plan name"
+  default     = ""
+}
+
+variable "target_vault_name" {
+  type        = string
+  description = "Override target Vault Name"
+  default     = ""
```
Comment (osterman) [inline variables.tf:80]: Use null instead?

File: variables.tf [hunk 8]
```diff
@@ -67,3 +67,39 @@ variable "copy_action_delete_after" {
   description = "For copy operation, specifies the number of days after creation that a recovery point is deleted. Must be 90 days greater than `copy_action_cold_storage_after`"
   default     = null
 }
+
+variable "plan_name_suffix" {
+  type        = string
+  description = "The string appended to the plan name"
+  default     = ""
```
Comment (osterman) [inline variables.tf:74]: Use null instead?

**Review by osterman — DISMISSED** [review]

File: variables.tf [hunk 9]
```diff
@@ -67,3 +67,39 @@ variable "copy_action_delete_after" {
...
+
+variable "iam_role_enabled" {
+  type        = bool
+  description = "Should we create a new Iam Role and Policy Attachment"
+  default     = true
+}
+
+variable "target_iam_role_name" {
+  type        = string
+  description = "Override target IAM Name"
```
Comment (aknysh) [inline variables.tf:103]:
```suggestion
  description = "Override target IAM Role Name"
```

**Review by osterman — APPROVED** [review]

**Review by osterman — APPROVED** [review]

**Pull request merged**